lib/Reader/reader.cpp

```diff
@@ -7865,12 +7862,7 @@ void ReaderBase::readBytesForFlowGraphNode(FlowGraphNode *Fg,
         false; // Blocks with errors can't have this verified
 
     // Delete all (non-EH reachability) flow edges that come from this block.
-    FlowGraphEdgeList *Arc, *ArcNext;
-    for (Arc = fgNodeGetSuccessorListActual(Fg); Arc != nullptr;
-         Arc = ArcNext) {
-      ArcNext = fgEdgeListGetNextSuccessorActual(Arc);
-      fgDeleteEdge(Arc);
-    }
+    fgRemoveAllActualSuccessorEdges(Fg);
```

Comment:
As you're freeing the base Reader class from its assumption that edges have objects, it might be nice to no longer need `DeleteEdge` (which is an awkward thing to try to implement on LLVM IR), mightn't it?  I notice that `ReaderBase::fgRemoveUnusedBlocks` is never called (only its override `ReaderIR::fgRemoveUnusedBlocks` is called), so you could remove that and then since it's the only caller of `fgDeleteBlockAndNodes` you could remove that and then since it in turn is the only caller of `fgDeleteNodesFromBlock` and `fgDeleteBlock` you could remove those as well.  Then if instead of adding `fgRemoveAllActualSuccessorEdges` (which bottoms out in NYI in `fgDeleteEdge`, you could move the NYI up here and remove `fgDeleteEdge` as well.
Feel free to defer/ignore if you don't want to scope-creep this change and/or don't agree that would be an improvement.